### PR TITLE
add support for response redirects

### PR DESCRIPTION
**Pull request by treywood**
as supported in jetbrains IDEs: https://jetbrains.com/help/rider/Exploring_HTTP_Syntax.html#response-redirect

### CODE REVIEW

File: grammar.js [hunk 1]
```diff
@@ -247,7 +256,7 @@ module.exports = grammar({
         _external_body: ($) =>
             seq(
                 $.external_body,
-                NL,
+                repeat1(NL),
```
Comment (treywood) [inline grammar.js:259]: kind of a drive-by fix: should be able to separate an external body from response handler scripts/response redirects with multiple newlines.

i.e. this should be valid, but isn't without `repeat1`:

```http
POST http://www.example.com
Content-Type: application/json

< ./body.json

> ./scripts/handler.json

>> response.json
```

**Comment by boltlessengineer** [comment]
This would be easier to implement now with recent updates in the grammar.
I’ll have a look if I can merge or rebase your commits later.
Merging PRs in tree-sitter parser is hard 😭

**Comment by treywood** [comment]
oof I actually think recent changes in main broke some thing outside of this PR. in particular, external bodies are broken with response handler scripts. specifically this test fails:

```http
GET https://www.example.com

< ./user.json

> /handler.json
```

expected result:
```query
(document
  (section
    (request
      method: (method)
      url: (target_url)
      body: (external_body
        path: (path))
      (res_handler_script
        (path))))
```

actual result:
```query
(document
  (section
    (request
      method: (method)
      url: (target_url)
      body: (raw_body)))
```

it seems like additional content after an external body causes everything to just be parsed as a raw body

**Comment by treywood** [comment]
@boltlessengineer I rebased on main, conflicts weren't _too_ terrible :)

**Comment by treywood** [comment]
@boltlessengineer bump :)

**Comment by boltlessengineer** [comment]
Merging now. Thank you for your contribution!
